pkg/cmd/pulumi/packagecmd/project_sdk_link.go

```diff
@@ -294,10 +296,17 @@ func linkPythonPackage(ws pkgWorkspace.Context, root string, pkg *schema.Package
 			return fmt.Errorf("error opening requirments.txt: %w", err)
 		}
 
-		fBytes = []byte(packageSpecifier + "\n" + string(fBytes))
-		err = os.WriteFile(fPath, fBytes, 0o600)
-		if err != nil {
-			return fmt.Errorf("could not write requirments: %w", err)
+		lines := regexp.MustCompile("\r?\n").Split(string(fBytes), -1)
+		if !slices.Contains(lines, packageSpecifier) {
+			if runtime.GOOS == "windows" {
+				fBytes = []byte(packageSpecifier + "\r\n" + string(fBytes))
```

Comment:
Does everyone on windows use `\r\n`?  I think we should base this on whether we already have `\r\n` in the file :thinking: 

Otherwise users collaborating across platforms are not gonna be happy.

Reply:
Right, how about this https://github.com/pulumi/pulumi/pull/18972/commits/e50314eca75f021df8fe7577f3d43bbb177a3d6f ?

Edge case if the file uses mixed endings, we'll use CRLF, which seems ok.

Reply:
Yeah, this looks good to me, thanks!  If someone uses mixed endings that's their problem tbh :sweat_smile: